utils.go

```diff
@@ -147,6 +148,32 @@ func parseDSN(dsn string) (cfg *config, err error) {
 		return nil, errInvalidDSNNoSlash
 	}
 
+	if cfg.interpolateParams && cfg.collation != defaultCollation {
+		// A whitelist of collations which safe to interpolate parameters.
+		// ASCII and latin-1 are safe since they are single byte encoding.
+		// utf-8 is safe since it doesn't conatins ASCII characters in trailing bytes.
+		safeCollations := []string{"ascii_", "latin1_", "utf8_", "utf8mb4_"}
```

Comment:
It might be better to use a blacklist instead. MySQL only supports 5 unsafe collations: `bgk`, `big5`, `cp932`, `gb2312` and `sjis`